libs/sdk-ui-dashboard/src/presentation/filterBar/attributeFilter/dashboardDropdownBody/configuration/hooks/useDependentDateFiltersConfiguration.ts

```diff
@@ -39,23 +36,47 @@ export function useDependentDateFiltersConfiguration(
         filterElementsByDate,
     );
 
+    const originalCommonDateFilterState = useDependentCommonDateFilterConfigurationState(commonDateFilter);
+
     const [dependentDateFilters, setDependentDateFilters] =
         useState<IDashboardDependentDateFilter[]>(originalState);
 
-    const onDependentDateFiltersSelect = (dataSet: ObjRef, isSelected: boolean) => {
-        invariant(!isUriRef(dataSet));
+    const [dependentCommonDateFilter, setDependentCommonDateFilter] = useState<IDashboardDateFilter>(
+        originalCommonDateFilterState,
+    );
+
+    const onDependentDateFiltersSelect = (
+        item: IDashboardDependentDateFilter,
+        isSelected: boolean,
+        isCommonDate?: boolean,
+    ) => {
+        const localIdentifier = item.dataSet;
+        invariant(!isUriRef(localIdentifier));
 
         const changedParentIndex = dependentDateFilters.findIndex(
-            (dependentDateFilter) => dependentDateFilter.localIdentifier === dataSet.identifier,
+            (dependentDateFilter) =>
+                dependentDateFilter.localIdentifier === localIdentifier?.identifier &&
+                dependentDateFilter.isCommonDate === isCommonDate,
```

Comment:
Isn't it possible the function would have ` isCommonDate` argument as `undefined` and the `dependentDateFilter.isCommonDate` would be either `true` or `false` and condition would not match? I find these conditions suspicious and possibly fragile. It would be better to have this `isCommonDate` non optional everywhere if that's possible or sanitize it somewhere to explicit boolean value.

Reply:
Make `isCommonDate` parameter as mandatory, so it properly checks it.